marketplace-service/src/main/java/com/axonivy/market/service/impl/FileDownloadServiceImpl.java

```diff
@@ -79,10 +71,11 @@ public String downloadAndUnzipFile(String url, boolean isForce) throws IOExcepti
     return location;
   }
 
-  private Path createTempFileFromUrlAndExtractToLocation(String url, String location,
+  public Path createTempFileFromUrlAndExtractToLocation(String url, String location,
       boolean isForce) throws IOException {
     File cacheFolder = new File(location);
-    if (cacheFolder.exists() && cacheFolder.isDirectory() && !isForce) {
+    if (cacheFolder.exists() && cacheFolder.isDirectory() && Objects.requireNonNull(
```

Comment:
If you put Objects.requireNonNull here. It will throws error instead of going to else condition 
in case of listFile return null